pkg/executor/bacalhau/bacalhau.go

```diff
@@ -26,221 +39,238 @@ type BacalhauExecutorOptions struct {
-			Disk: int(node.Info.ComputeNodeInfo.MaxCapacity.Disk),
-		}
-		for _, gpu := range node.Info.ComputeNodeInfo.MaxCapacity.GPUs {
-			spec.GPUs = append(spec.GPUs, data.GPUSpec{
-				Name:   gpu.Name,
-				Vendor: string(gpu.Vendor),
-				VRAM:   int(gpu.Memory),
-			})
-		}
-		specs = append(specs, spec)
-	}
-	return specs, nil
+	return executor.bacalhauClient.GetMachineSpecs()
 }
 
 func (executor *BacalhauExecutor) RunJob(
 	deal data.DealContainer,
 	module data.Module,
 ) (*executorlib.ExecutorResults, error) {
-	id, err := executor.getJobID(deal, module)
+	fmt.Printf("Running job: %s\n", deal.ID)
+	jobId, err := executor.getJobID(deal, module)
 	if err != nil {
 		return nil, err
 	}
 
-	jobState, err := executor.getJobState(deal.ID, id)
-	if err != nil {
-		return nil, err
-	}
+	var jobExecutions []*models.Execution
+	for {
+		jobInfo, err := executor.bacalhauClient.GetJob(jobId)
+		if err != nil {
+			return nil, fmt.Errorf("error getting job %s: %s", jobId, err.Error())
+		}
 
-	if len(jobState.State.Executions) <= 0 {
-		return nil, fmt.Errorf("no executions found for job %s", id)
-	}
+		if jobInfo.Executions == nil {
+			return nil, fmt.Errorf("no executions retrieved for job %s", jobId)
+		}
+
+		jobExecutions = jobInfo.Executions.Items
+
+		if len(jobExecutions) > 0 {
+			if jobInfo.Job.State.StateType != models.JobStateTypeCompleted {
+				return nil, fmt.Errorf("job %s did not complete yet: %s", jobId, jobInfo.Job.State.StateType.String())
+			}
+			break
+		}
 
-	// Check that the job completed successfully
-	if jobState.State.State != bacalhau.JobStateCompleted {
-		return nil, fmt.Errorf("job %s did not complete successfully: %s", id, jobState.State.State.String())
+		time.Sleep(5 * time.Second)
 	}
 
 	system.EnsureDataDir(RESULTS_DIR)
 	resultsDir := system.GetDataDir(filepath.Join(RESULTS_DIR, deal.ID))
-	cidString := jobState.State.Executions[0].PublishedResult.CID
-	err = executor.ipfsClient.Get(context.Background(), cidString, resultsDir)
+
+
+	cid,resultsDir, err := executor.prepareResults(jobId,jobExecutions[0].ID)
 	if err != nil {
-		return nil, fmt.Errorf("error getting results from IPFS %s -> %s", deal.ID, err)
+		return nil, fmt.Errorf("error preparing results: %s", err.Error())
 	}
 
-	// TODO: we should think about WASM and instruction count here
 	results := &executorlib.ExecutorResults{
 		ResultsDir:       resultsDir,
-		ResultsCID:       jobState.State.Executions[0].PublishedResult.CID,
+		ResultsCID:       cid,
 		InstructionCount: 1,
 	}
 
 	return results, nil
 }
 
-// run the bacalhau job and return the job ID
-func (executor *BacalhauExecutor) getJobID(
-	deal data.DealContainer,
-	module data.Module,
-) (string, error) {
-	// get a JSON string of the job
-	jsonBytes, err := json.Marshal(module.Job)
+func (executor *BacalhauExecutor) prepareResults(jobId string, executionId string) (cid string, resultsDir string, err error) {
+	
+	home, err := os.UserHomeDir()
```

Comment:
Why are we using the user's home directory? We already had `RESULTS_DIR` and `EnsureDataDir`. I don't understand this change